Review history

pollyv opened the pull request:
- При нажатии на «стрелку» открывается сайдбар с настройками.
- При повторном нажатии на «стрелку», клике вне сайдбара или нажатии на Esc сайдбар закрывается.
- После нажатия на «Применить» стили применяются к статье.
- При нажатии «Сбросить» настройки в форме сбрасываются на начальные, которые были при открытии страницы.

pollyv committed:
First review: реализованы основные требования
- При нажатии на «стрелку» открывается сайдбар с настройками.
- При повторном нажатии на «стрелку», клике вне сайдбара или нажатии на Esc сайдбар закрывается.
- После нажатия на «Применить» стили применяются к статье.
- При нажатии «Сбросить» настройки в форме сбрасываются на начальные, которые были при открытии страницы.

mikebars1995 commented on src/index.tsx:
```diff
@@ -9,28 +9,39 @@ import { defaultArticleState } from './constants/articleProps';
 const root = createRoot(domNode);
 
+// Главный компонент
 const App = () => {
```
**Можно лучше**


* Лучше выделить компонент  `App` в отдельный файл, чтобы не смешивать все в `index.js`

mikebars1995 commented on src/index.tsx:
```diff
@@ -9,28 +9,39 @@ import { defaultArticleState } from './constants/articleProps';
+	const [defaultArticle, setDefaultArticle] = useState(defaultArticleState);
+
 	return (
 		<div
```
**Можно лучше**

 * По семантике нужно помещать весь главный контент страницы **после** хэдера в тег `main`.

mikebars1995 commented on src/index.tsx:
```diff
@@ -9,28 +9,39 @@ import { defaultArticleState } from './constants/articleProps';
+
 	return (
 		<div
 			className={clsx(styles.main)}
```
**Можно лучше**


* `clsx` можно не использовать, если только **один** селектор передается

mikebars1995 commented on src/components/article-params-form/ArticleParamsForm.tsx:
```diff
@@ -1,14 +1,160 @@
+
+	// Обработчик закрытия формы, если нажимаем Escape
+	useEffect(() => {
+		function closeForm(event: KeyboardEvent) {
```
**Нужно исправить**


* Нужно **останавливать эффект**, если форма закрыта, чтобы не навешивать обработчик 


```js
if (!isOpen) return;
```

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


Что сделано отлично:

* Хорошо реализовали форму из предоставленных компонентов
* Хорошая логика передачи данных от `App` в форму и обратно
* Отлично, что обработчик сабмита вешаете на форму, а не клик на кнопку. 
* Отлично, что нет типов `any`
* Отлично типизировали проект

Что нужно исправить:



* Нужно **останавливать эффект**, если форма закрыта, чтобы не навешивать обработчик 




Что можно улучшить:

* По семантике нужно помещать весь главный контент страницы **после** хэдера в тег `main`.
* Лучше выделить компонент  `App` в отдельный файл, чтобы не смешивать все в `index.js`
* `clsx` можно не использовать, если только **один** селектор передается

Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода.		 

pollyv committed:
Second review: исправлены критические ошибки
- Эффект остановлен, если форма закрыта.
- Весь главный контент страницы помещен в тег main.
- Компонент App вынесен в отдельный файл.
- clsx можно не используется, если только один селектор передается.

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
